src/runtime_src/core/tools/common/XBHelpMenus.cpp

```diff
@@ -266,31 +266,34 @@ XBUtilities::produce_reports( xrt_core::device_collection devices,
         is_recovery = false;
       }
 
+      // Process the tests that require a device
+      // If the device is either of the following, most tests cannot be completed fully:
+      // 1. Is in factory mode and is not in recovery mode
+      // 2. Is not ready and is not in recovery mode
+      if((is_mfg || !is_ready) && !is_recovery) {
+        std::cerr << "Error: Device is not ready - Limited functionality available with XRT tools.\n\n";
+        is_report_output_valid = false;
+      }
+
       for (auto &report : reportsToProcess) {
         if (report->isDeviceRequired() == false)
           continue;
-        //if the device is either of the following, continue to create reports:
-        // 1. not in factory mode and is ready to use
-        // 2. is in recovery mode
-        if((!is_mfg && !is_ready) && !is_recovery)
-          continue;
+
         boost::property_tree::ptree ptReport;
         try {
           report->getFormattedReport(device.get(), schemaVersion, elementFilter, consoleStream, ptReport);
         } catch (const std::exception&) {
           is_report_output_valid = false;

```

Comment:
Would outputting the exception message here be a good idea? It is masking the error from getFormattedReport

Reply:
If we were at the start of the release, I would say yes, but we are instead at the end so we need to keep our changes to a minimum.   

Please take a note of this request and look into post 2022.1 how this might be done cleanly.